internal/testdata/dlaqr5test/main.go

```diff
@@ -0,0 +1,216 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This program generates test data for Dlaqr5. Test cases are stored as text
+// files inside dlaqr5data.zip archive which is then read by testlapack/dlaqr5.go.
+//
+// This program uses cgo to call Fortran version of DLAQR5. Therefore, matrices
+// generated by hessrand and eye are in column-major format but are written into
+// test case files in row-major format. See writeCase and writeCaseWant for
+// details.
+package main
+
+// void dlaqr5_(int* wantt, int* wantz, int* kacc22, int* n, int* ktop, int* kbot, int* nshfts,
+//              double* sr, double* si, double* h, int* ldh, int* iloz, int* ihiz,
+//              double* z, int* ldz, double* v, int* ldv, double* u, int* ldu,
+//              int* nv, double* wv, int* ldwv, int* nh, double* wh, int* ldwh);
+import "C"
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"log"
+	"math/rand"
+	"os"
+)
+
+func fortranDlaqr5(wantt, wantz bool, kacc22 int, n, ktop, kbot int, nshfts int, sr, si []float64, h []float64,
+	ldh int, iloz, ihiz int, z []float64, ldz int, v []float64, ldv int,
+	u []float64, ldu int, nh int, wh []float64, ldwh int, nv int, wv []float64, ldwv int) {
+	func() {
+		wt := C.int(0)
+		if wantt {
+			wt = 1
+		}
+		wz := C.int(0)
+		if wantz {
+			wz = 1
+		}
+		kacc22 := C.int(kacc22)
+		n := C.int(n)
+		ktop := C.int(ktop + 1)
+		kbot := C.int(kbot + 1)
+		nshfts := C.int(nshfts)
+		ldh := C.int(ldh)
+		iloz := C.int(iloz + 1)
+		ihiz := C.int(ihiz + 1)
+		ldz := C.int(ldz)
+		ldv := C.int(ldv)
+		ldu := C.int(ldu)
+		nh := C.int(nh)
+		ldwh := C.int(ldwh)
+		nv := C.int(nv)
+		ldwv := C.int(ldwv)
+		C.dlaqr5_((*C.int)(&wt), (*C.int)(&wz), (*C.int)(&kacc22),
+			(*C.int)(&n), (*C.int)(&ktop), (*C.int)(&kbot),
+			(*C.int)(&nshfts), (*C.double)(&sr[0]), (*C.double)(&si[0]),
+			(*C.double)(&h[0]), (*C.int)(&ldh),
+			(*C.int)(&iloz), (*C.int)(&ihiz), (*C.double)(&z[0]), (*C.int)(&ldz),
+			(*C.double)(&v[0]), (*C.int)(&ldv),
+			(*C.double)(&u[0]), (*C.int)(&ldu),
+			(*C.int)(&nh), (*C.double)(&wh[0]), (*C.int)(&ldwh),
+			(*C.int)(&nv), (*C.double)(&wv[0]), (*C.int)(&ldwv))
+	}()
+}
+
+func main() {
+	const tmpl = "wantt%v_n%v_nshfts%v_ktop%v_kbot%v.txt"
+
+	file, err := os.Create("dlaqr5data.zip")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+	zipfile := zip.NewWriter(file)
+
+	rnd := rand.New(rand.NewSource(1))
```

Comment:
I added this common random generator to get higher variability in generated matrices. Indeed, it increases coverage by taking the branch at dlaqr5.go:416. It also required the equality tolerance to be increased to 1e-13.